データベース設計 README.mkの作成

yufuworks opened the pull request:
# What?
データベース設計の仕様書です。
# Why?
開発開始前に当初のデータベース仕様の策定、
サイトの構成を確定するため。

yufuworks committed:
EditReadmeFile

yufuworks committed:
Fixed filetype and add text format

最初にアップしたファイル形式のミスを訂正、および設計書式フォーマットを書き込み。

yufuworks committed:
Databese design has done

plz, check it

katsu105 commented on config/README.md:
```diff
@@ -0,0 +1,85 @@
+|image|string|null] false|
+
+### Association
+has_one :comment
```
imagesテーブルと画像を分けている理由はありますか？
一緒にしてしまってもいいかと思います。

yufuworks replied:
基礎カリキュラムで作成したものでavatarというgemを用いて画像アップロードするものがありました。その時、userとhas_one関係で拡張された形になったのでgemを用いて特定機能もったテーブル拡張する場合、テーブルが別れる可能性がある、というのが一点。
そうでなくてもchatspaceにおいて将来的に１回のコメントに複数画像を貼れるようにする場合のような機能拡張を考えると画像テーブルは別れていると良いのかな、と思ったのが一点です。

katsu105 replied:
>chatspaceにおいて将来的に１回のコメントに複数画像を貼れるようにする場合のような機能拡張を考えると

拡張性を考えているところが非常に素晴らしいです！
実際の実装ではそうする可能性も十分に考えられます。

ただ、今回はchatspaceというカリキュラムで決められたアプリケーションであり、カリキュラムに沿った形で完成していただくことが目的となりますので、commentsテーブルにまとめていただくことは可能でしょうか？

yufuworks replied:
了解しました。
書き直してアップロードします！

yufuworks replied:
修正しました

katsu105 commented on config/README.md:
```diff
@@ -0,0 +1,85 @@
+|name|string|null: false, default: false|
+
+### Association
+has_many :users, through: :groups_users
```
アソシエーションが足りないので記述しましょう。

yufuworks replied:
確認して修正します

yufuworks replied:
修正しました

katsu105 replied:
commentsテーブルに対してもアソシエーションが必要そうです。

yufuworks replied:
修正しました

katsu105 commented on config/README.md:
```diff
@@ -0,0 +1,85 @@
+### Association
+has_many :groups, through: :groups_users
+has_many :comments
+has_many :images
```
アソシエーションが足りないので記述しましょう。

yufuworks replied:
確認して修正します

yufuworks replied:
修正しました

katsu105 reviewed (approved):
アソシエーションが不足しているので追記してください。

yufuworks committed:
Fixed associations

add sentenceis to junction table for fix those

yufuworks committed:
Fixed Table assets

Table構造の見直し => imagesテーブル削除

これに伴い下記の記述を変更しました。
・usersテーブル、commentテーブルのアソシーエション
・commentsテーブルのimage_id属性および値の変更

yufuworks committed:
Fixed markdown syntax

マークダウンの記述エラーでindex設計に関する記述が表示されていなかったエラーを修正しました。

※前二回までのpushにて
アソシエーションの記述もれ修正
および、table構成の変更をしています。

katsu105 commented on config/README.md:
```diff
@@ -0,0 +1,70 @@
+|------|----|-------|
+|id|ingeger|null: false|
+|nickname|string|null: false|
+|email|string|null: false|
```
emailカラムはdeviseで自動生成されるので削除してしまって構いません。

yufuworks replied:
修正しました！

yufuworks replied:
この件について質問ですが、gemなどの外部機能によってテーブルを作成する前提の場合、
この仕様書にはどのgemを使う、などのようなメモを（コメントアウトせずに）記載するべきでしょうか？
当たり前すぎる前提で通じるものかもしれませんが、可読性の観点ではそのような特記事項が必要なのではないか、と感じます。
よろしくお願いいたします。

katsu105 commented on config/README.md:
```diff
@@ -0,0 +1,70 @@
+TODO:
+user削除時の挙動、投稿済みのコメント（イメージ）は残してnull時のデフォルトname表示が必要？
+削除時のnull専用のuserを予め登録しておく？要調査だがひとまず保留、あとで修正対応かも。
+ -->
```
こちらのコメントアウトは削除しましょう。

yufuworks replied:
削除しました

katsu105 commented on config/README.md:
```diff
@@ -0,0 +1,70 @@
+グループ作成userのmastar判定をtrueとし権限を与える
+フォームから追加されたメンバーにはデフォルト設定のfalseを入れる
+もし必要ないなら（全員mastarにするなどの場合）mastarカラムは後々削除
+ -->
```
こちらのコメントアウトは削除しましょう。

yufuworks replied:
削除しました

katsu105 commented on config/README.md:
```diff
@@ -0,0 +1,70 @@
+|------|----|-------|
+|user_id|integer|null: false, foreign_key: true|
+|group_id|integer|null: false, foreign_key: true|
+|master|bool|null: false|
```
こちらのmasterカラムは何を指していますか？

yufuworks replied:
groupの編集権利のスイッチが必要かと思いました。
最初にgroupを作成したuserにtrue,以後追加されるuserをデフォルトでfalseとして
viewで編集ボタンのon,offができるのではないかな？と。

katsu105 replied:
>groupの編集権利のスイッチが必要かと思いました。

そこまで考えられている視野の広さが非常に良いですね！
ただ今回はそういった実装内容はないので削除していただけると幸いです。

yufuworks replied:
了解しました、削除いたします！

katsu105 commented on config/README.md:
```diff
@@ -0,0 +1,70 @@
+<!-- 
+userを削除した時紐付けがどうなるのか。
+通常表示するuser.nameをどうすれば良いか。
+ -->
```
こちらのコメントアウトは削除しましょう。

yufuworks replied:
削除しました

katsu105 commented on config/README.md:
```diff
@@ -0,0 +1,70 @@
+|image|string|null: ture|
+
+### Association
+belongs_to :users
```
groupに対してもアソシエーションが必要そうです。

yufuworks replied:
group-comment間のアソシエーションを追記しました。

katsu105 reviewed (commented):
idカラムは自動的に生成されるので記述する必要はないです。削除しましょう。

yufuworks commented:
すいません、別に作ってたブランチの方でpushしてました、pushし直します。

yufuworks committed:
Add association, delete id columns, commentouts

・group-comment間のアソシエーションを追記
・コメントアウトを全て削除しました。
・各テーブルからidカラムを削除
・userテーブルから自動生成されるemailの記述を削除

yufuworks committed:
Deleted groups_users from master column

・groups_usersテーブルからmasterカラムを削除

※別ブランチで一度間違えてREADME変更をpushしたためローカル環境が混乱しましたがこのpushでコメント削除なども反映されているはず、です。

ご確認お願いします。

katsu105 commented:
LGTMです。お疲れ様でした🎉

yufuworks committed:
Add group to foreign key

・テーブル構成変更時に落としていた外部キーをcommentsテーブルに追加修正しました。
・別ブランチpush修正時に消えていたcommentsテーブルのアソシエーションを修正しました。

yufuworks commented:
READMEファイルにおいて外部キーの記述もれ、アソシエーションの変更をしています。
←https://github.com/yufuworks/chat-space/pull/1

Gemfileの追記＞ブランチ間違いでREADMEプッシュ＞revertプッシュ
⇦https://github.com/yufuworks/chat-space/pull/2

状況が入り乱れております、お手数ですが改めてご確認いただきたく思います。

ブランチ並行からのpush間違いでローカル側のブランチ間にある同一ファイルの扱いを間違えるととても面倒なことになること、それに付随してgithub上でのコミュニケーションにも混乱が起こることがよくわかりました。この操作ミスはかなりありがちにやらかしそうです。可能な限りブランチ並行作業させない方がいいことがよくわかりました。大変お手間とらせてしまいましたが、むしろ初期のこの段階で経験できて良かったということにさせてください🙇‍♂️

katsu105 commented:
良い経験だと感じていただき幸いです。
ブランチの並行作業は今後ある可能性が高いので、git stashやgit resetなどのコマンドを覚えといた方が役に立つと思います。
お疲れ様でした！
